src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,404 @@
+        hsvCheckButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_HSV_BUTTON_TOOLTIP");
+        rawCheckButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_RAW_BUTTON_TOOLTIP");
+
+        if (HsvMode)
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_H_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_S_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_V_TOOLTIP");
+        }
+        else
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_R_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_G_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_B_TOOLTIP");
+        }
+
+        sliderA.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_A_TOOLTIP");
+    }
+
+    private void UpdateButtonsState()
+    {
+        if (hsvCheckButton == null)
+            return;
+
+        hsvCheckButton.Pressed = HsvMode;
+        rawCheckButton.Pressed = RawMode;
+
+        hsvCheckButton.Disabled = !hsvButtonEnabled || RawMode;
+        rawCheckButton.Disabled = !rawButtonEnabled || HsvMode;
+    }
+
+    private void OnAddPresetButtonPressed()
+    {
+        if (!presetsEnabled)
+            return;
+
+        AddPreset(Color);
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        HsvMode = isOn;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        RawMode = isOn;
+    }
+
+    private void OnColourChanged(Color colour)
+    {
+        htmlColourEdit.Text = colour.ToHtml();
+    }
+
+    /// <summary>
+    ///   Called when (keyboard) entered in HtmlColourEdit or from OnHtmlColourEditFocusExited.
+    ///   Set Color when text is valid; reset if not.
+    /// </summary>
+    /// <param name="colour">Current htmlColourEditor text</param>
+    private void OnHtmlColourEditEntered(string colour)
+    {
+        if (colour.IsValidHtmlColor())
+        {
+            Color = new Color(colour);
+        }
+        else
+        {
+            htmlColourEdit.Text = Color.ToHtml();
+        }
+    }
+
+    /// <summary>
+    ///   Called when focus exited HtmlColourEdit.
+    /// </summary>
+    private void OnHtmlColourEditFocusExited()
+    {
+        OnHtmlColourEditEntered(htmlColourEdit.Text);
+        htmlColourEdit.Deselect();
+    }
+
+    private class TweakedColourPickerPreset : ColorRect
+    {
+        private readonly TweakedColourPicker owner;
+
+        public TweakedColourPickerPreset(TweakedColourPicker owner, Color colour)
+        {
+            this.owner = owner;
+            Color = colour;
+            MarginTop = MarginBottom = MarginLeft = MarginRight = 6.0f;
+            RectMinSize = new Vector2(20, 20);
+            SizeFlagsHorizontal = 8;
+            SizeFlagsVertical = 4;
```

Comment:
Aren't there named enums for these?
Looks like there are, `SizeFlags.ShrinkCenter` is a thing that exists. So those should be used instead of magic numbers.